base/gfsppieceop/pieceop.go

```diff
@@ -62,3 +66,17 @@ func (p *GfSpPieceOp) SegmentPieceCount(payloadSize uint64, maxSegmentSize uint6
 	}
 	return uint32(count)
 }
+
+func (p *GfSpPieceOp) ParseSegmentIdx(segmentKey string) (uint32, error) {
+	segments := strings.Split(segmentKey, "_")
+	if len(segments) != 2 {
+		return 0, fmt.Errorf("invalid segmentKey format")
+	}
+
+	segmentIdx, err := strconv.ParseUint(segments[1][1:], 10, 32)
```

Comment:
what does  `[1:]` mean. Sse some temporary variables to make the code clearer.

Reply:
for example, the object name is " s123_s4",  the segments[1] is  "s4 ", and  the index should be s4[1:].

Reply:
changed to 	```
segPrefix := "s"
	segmentIdx, err := strconv.ParseUint(segments[1][len(segPrefix):], 10, 32)
```